File: solutions/swift/raindrops/1/Sources/Raindrops/Raindrops.swift
let divisors = [3, 5, 7]
let sounds = ["Pling", "Plang", "Plong"]
let drops = zip(divisors, sounds)

func raindrops(_ number: Int) -> String {
    let result = drops.reduce(into: "") { result, pair in
        let (divisor, sound) = pair
        if number.isMultiple(of: divisor) {
            result += sound
        }
    }
    return result.isEmpty ? String(number) : result
}

File: solutions/elixir/raindrops/6/lib/raindrops.ex
defmodule Raindrops do

  def convert(number) do
    drop = [
      &if(rem(&1, 3) == 0, do: "Pling", else: ""),
      &if(rem(&1, 5) == 0, do: "Plang", else: ""),
      &if(rem(&1, 7) == 0, do: "Plong", else: "")
    ]
    |> Enum.map(& &1.(number))
    |> Enum.join()

    if drop == "", do: to_string(number), else: drop
  end

end
